prov/shm/src/smr_progress.c

```diff
@@ -1062,18 +1149,65 @@ static void smr_progress_cmd(struct smr_ep *ep)
 	}
 }
 
+static void smr_progress_ipc_list(struct smr_ep *ep)
+{
+	struct smr_pend_entry *ipc_entry;
+	struct smr_domain *domain;
+	struct dlist_entry *tmp;
+	int ret;
+
+	domain = container_of(ep->util_ep.domain, struct smr_domain,
+			      util_domain);
+
+	pthread_spin_lock(&ep->region->lock);
```

Comment:
Do we need to lock the whole region for this? Can we just use the util_ep lock like the sar processing does?